comms/discovery/rpcz/apply.go

```diff
@@ -462,6 +462,8 @@ func websocketNotify(rpcJson json.RawMessage, userId int32, timestamp time.Time)
 			websocketPush(subscribedUserId, j)
 		}
 
+	} else if gjson.GetBytes(rpcJson, "method").String() == "chat.blast" {
+		websocketPushAll(rpcJson, timestamp)
```

Comment:
This feels like it should be identical to the above except the `userIds` list comes from the current list of websockets not from the db? Why not use the same function in both places w/ a userIds param, or have `websocketPushAll` use `websocketPush`? There seems like there should be some reuse here

Reply:
I guess `websocketPush` has the side effect of potentially sending the blast upgrade message multiple times within 10 seconds?

Reply:
the thing i was trying to avoid was having this code reach into the websocket and grab the userId, felt like bad coupling. but i didn't think of `websocketPushAll` just calling `websocketPush`, that could work. we'd still have to form the metadata inside `websocketPushAll` but that's probably ok.

not sure what you're talking about with the 10 seconds thing?

Reply:
The code in `websocketPush` caches the recent messages for up to 10 seconds and resends them on each new message it looks like

Reply:
ah i see it. i think that's still fine tho? that only happens if the websocket gets disconnected/reconnected right? and if multiple blast rpcs come in for the same blast message, the worst that'll happen is it'll trigger multiple calls to `getBlasts()` but the data should all be fine since we upsert/don't insert dups etc.

Reply:
hm one downside to `websocketPushAll` calling `websocketPush` is that we'll iterate through the websocket list `n^2` times... probably not the end of the world but not great.. dang should i favor perf or code re-use??